homeassistant/components/fyta/coordinator.py

```diff
@@ -0,0 +1,64 @@
+"""Coordinator for FYTA integration."""
+
+from datetime import datetime, timedelta
+import logging
+
+from fyta_cli.fyta_connector import FytaConnector
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FytaCoordinator(DataUpdateCoordinator):
+    """Fyta custom coordinator."""
+
+    def __init__(
+        self, hass: HomeAssistant, fyta: FytaConnector, entry: ConfigEntry
+    ) -> None:
+        """Initialize my coordinator."""
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="FYTA Coordinator",
+            update_method=self._async_update_data,
+            update_interval=timedelta(seconds=60),
+        )
+
+        self.fyta = fyta
+        self.config_entry: ConfigEntry = entry
+
+        self.plant_list: dict[int, str] = {}
+        self.access_token = ""
+        self.expiration = None
+        self._attr_last_update_success = None
+
+    async def _async_update_data(self):
+        """Fetch data from API endpoint."""
+
+        if self.access_token == "" or self.expiration < datetime.now():
+            await self.renew_authentication()
+
+        data = await self.fyta.update_all_plants()
+        data |= {"online": True}
+
+        self.plant_list = self.fyta.plant_list
+        data |= {"plant_number": len(self.plant_list)}
+        data |= {"email": self.fyta.email}
+        data |= {"name": "Fyta Coordinator"}
+
+        self._attr_last_update_success = datetime.now()
+        return data
```

Comment:
I have now deleted "name" and "online" (the latter I wanted to use for binary sensor, that is now excluded). Plant_number is a sensor of the coordinator and email is required to get the Device Name.

Reply:
This online shouldn't be used for a binary sensor. If you had an unsuccessful update you should raise UpdateFailed, which causes the entities that use this coordinator to go unavailable. 

The plant number is literally checking the length of the data, so there's no point in already doing that here since you already can return the data.

Reply:
I have removed plant_list as well. However, it then becomes more complicated to create the sensors in sensor.py. With the above, we can just go over the keys in coordinator.data. If it is deleted, we need to add a special routine to get the FytaSensorEntityDescription for plant_list and add this separately. In my view it would be easier to keep plant_list in data, what are your thoughts?

Reply:
I guess I was thinking too complicated and thought I had to add a new comprehension just for plant_number. I now extended the comprehension as follows:
```
    plant_entities = (
        [
            FytaCoordinatorSensor(coordinator, entry, sensor)
            for sensor in SENSORS
            if sensor.key in coordinator.data or sensor.key == "plant_number"
        ]
    )
```